oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java

```diff
@@ -229,6 +271,11 @@ public static String getServiceAccountsUrl() {
         + "/computeMetadata/v1/instance/service-accounts/?recursive=true";
   }
 
+  public static String getIdentityDocumentUrl() {
+    return getMetadataServerUrl(DefaultCredentialsProvider.DEFAULT)
+        + "/computeMetadata/v1/instance/service-accounts/default/identity";
+  }
+
   @Override
   public int hashCode() {
     return Objects.hash(transportFactoryClassName);
```

Comment:
not for this PR, but this looks really wonky. Note to myself: can this possibly be correct?